src/dspeed/processing_chain.py

```diff
@@ -1407,12 +1407,21 @@ def __init__(
 
                 # reshape just in case there are some missing dimensions
                 arshape = list(param.shape)
-                for idim in range(-1, -1 - len(shape), -1):
-                    if (
-                        len(arshape) < len(shape) + 1 + idim
-                        or arshape[idim] != shape[idim]
-                    ):
-                        arshape.insert(len(arshape) + idim + 1, 1)
+                # check of shape mismatch is just due to wf buffer dim being included
```

Comment:
I suggest reverting this change, and instead adding after line 1263:
`if not param.is_const: arr_dims += [self.proc_chain._bock_width]`

Reply:
This doesn't seem to work in testing as the arshape doesn't have the block_width either

Reply:
so it breaks it for all the other processors

Reply:
Oh yeah, the block width dim has to go first, try `arr_dims = [self.proc_chain._bock_width] + arr_dims` instead